### PR TITLE
feat: support system tag

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
/kind feature

**What this PR does / why we need it**:

| Tags | SystemTags | existing tags on resources | new tags on resources |
| ----- | ------------ | ----- | ----- |
| "a=b,c=d" | "" | {} | {"a": "b", "c": "d"} |
| "a=b,c=d" | "" | {"a": "x", "c": "y"} | {"a": "b", "c": "d"} |
| "a=b,c=d" | "" | {"e": "f"} | {"a": "b", "c": "d", "e": "f"} /* won't delete `e` because the SystemTags is empty */ |
| "c=d" | "a" | {"a": "b"} | {"a": "b", "c": "d"} /* won't delete `a` because it's in the SystemTags */ |
| "c=d" | "x" | {"a": "b"} | {"c": "d"} /* will delete `a` because it's not in Tags or SystemTags */ |

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-sigs/cloud-provider-azure/issues/499

**Special notes for your reviewer**:

**Release note**:
```
feat: support system tag
```

### CODE REVIEW

**Comment by nilo19** [comment]
would enrich the doc after this one is merged

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -2289,12 +2290,32 @@ func (az *Cloud) ensurePIPTagged(service *v1.Service, pip *network.PublicIPAddre
 	if serviceNames != nil {
 		configTags[consts.ServiceTagKey] = serviceNames
 	}
+
+	var systemTags []string
+	var systemTagsMap sets.String
+	if az.SystemTags != "" {
+		systemTags = strings.Split(az.SystemTags, consts.TagsDelimiter)
```
Comment (feiskyer) [inline pkg/provider/azure_loadbalancer.go:2297]: nit: delete all the spaces

File: pkg/provider/azure_loadbalancer.go [hunk 2]
```diff
@@ -2289,12 +2290,32 @@ func (az *Cloud) ensurePIPTagged(service *v1.Service, pip *network.PublicIPAddre
...
+
+	// if the systemTags is not set, just add/update new tags and not delete old tags
 	for k, v := range configTags {
 		if vv, ok := pip.Tags[k]; !ok || !strings.EqualFold(to.String(v), to.String(vv)) {
 			pip.Tags[k] = v
 			changed = true
 		}
 	}
+
+	// if the systemTags is set, delete the old tags
```
Comment (feiskyer) [inline pkg/provider/azure_loadbalancer.go:2309]: add a function for this since it is referenced by multiple places

Comment (nilo19) [reply pkg/provider/azure_loadbalancer.go]: done

File: pkg/provider/azure_loadbalancer.go (hunk 1 above)
Comment (nilo19) [reply pkg/provider/azure_loadbalancer.go]: done

File: pkg/provider/azure_loadbalancer.go (hunk 2 above)
Comment (nilo19) [reply pkg/provider/azure_loadbalancer.go]: done

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**